src/js/util/ui.js

```diff
@@ -45,5 +53,53 @@ define(function (require, exports) {
         });
     };
 
+    /**
+     * Asynchronously get the basic list of hit layer IDs in given document
+     *
+     * @param {number} id Document ID
+     * @param {number} x Horizontal coordinate
+     * @param {number} y Vertical coordinate
+     * @return {Promise.<Immutable.List<number>>}
+     */
+    var hitTestLayers = function (id, x, y) {
+        var documentRef = documentLib.referenceBy.id(id),
+            hitPlayObj = hitTestLib.layerIDsAtPoint(documentRef, x, y);
+
+        return descriptor.playObject(hitPlayObj)
+            .get("layersHit")
+            .then(function (ids) {
+                return Immutable.List(ids);
+            }, function () {
+                return Immutable.List();
+            });
+    };
+
+    /**
+     * Asynchronously get the color at the pixel of given coordinates in the document
+     *
+     * @param {number} id Document ID 
+     * @param {number} x Horizontal coordinate
+     * @param {number} y Vertical coordinate
+     * @param {number?} opacity Opacity of the owner layer at the pixel
+     * @return {Promise.<Color>}
+     */
+    var colorAtPoint = function (id, x, y, opacity) {
+        var documentRef = documentLib.referenceBy.id(id),
+            hitPlayObj = hitTestLib.colorSampleAtPoint(documentRef, Math.round(x), Math.round(y));
+
+        opacity = opacity || 100;
```

Comment:
This would convert `opacity === 0` into `opacity === 100`. You should declare `@param {number=}` and then test `opacity === undefined`.